### PR TITLE
Various tweaks to the kernel extension

### PR DESCRIPTION
**Pull request by fingolfin**
Please let me know if I should rebase this on some other branch instead of master (say, stable-0.13 ?)

### CODE REVIEW

File: src/digraphs.c [hunk 1]
```diff
@@ -25,6 +25,14 @@
 static Obj FuncDIGRAPH_OUT_NBS(Obj self, Obj digraph, Obj source, Obj range);
 static Obj FuncDIGRAPH_SOURCE_RANGE(Obj self, Obj digraph);
 
+#if !defined(GAP_KERNEL_MAJOR_VERSION) || GAP_KERNEL_MAJOR_VERSION < 3
+// compatibility with GAP <= 4.9
+static inline Obj NEW_PLIST_IMM(UInt type, Int plen)
+{
```
Comment (james-d-mitchell) [inline src/digraphs.c]: Brace at the end of the previous line please @fingolfin

Comment (fingolfin) [reply src/digraphs.c]: Oops, indeed! Will fix and rebase on stable-0.13

**Comment by wilfwilson** [comment]
Part of the problem with Travis was that `TestDirectory` now returns lines like
```
              0 failures in 4 files
```
which our systems parsed as a failure because the string fail appeared (change introduced here: https://github.com/gap-system/gap/commit/1cd18b77b986b6d4f01cadd9f196dd57fd08876d). I've fixed this by not grepping for `fail' at the end.

For the failing coverage job, this occurred when running a test file (in this instance `tst/standard/attr.tst`) with the required version of Orb, namely 4.7.5. Running
```
echo "Test(\"tst/standard/attr.tst\");" | ../../bin/gap.sh -A
```
from the digraphs directory would give
```
# line 437 of 1698 (25%)dyld: lazy symbol binding failed: Symbol not found: _FuncPermList
  Referenced from: /Users/Wilf/gap/pkg/orb/bin/x86_64-apple-darwin17.7.0-default64-kv6/orb.so
  Expected in: flat namespace

dyld: Symbol not found: _FuncPermList
  Referenced from: /Users/Wilf/gap/pkg/orb/bin/x86_64-apple-darwin17.7.0-default64-kv6/orb.so
  Expected in: flat namespace

Abort trap: 6
```
This doesn't happen with Orb 4.7.6 so I increased the required version of Orb to 4.7.6.

**Comment by wilfwilson** [comment]
@fingolfin If you please rebase one last time onto `stable-0.13` then everything should pass, and we'll merge.

**Comment by wilfwilson** [comment]
Thanks @fingolfin

**Pull request merged**